### PR TITLE
[yomm2] Update to v1.4.0

### PR DESCRIPTION
**Pull request by FabienPean**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/yomm2/fix_find_boost.patch [hunk 1]
```diff
@@ -20,7 +20,18 @@ index 7272504..6423a17 100644
...
  include("${CMAKE_CURRENT_LIST_DIR}/YOMM2Targets.cmake")
+diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
+index 7e8a282..f550c2d 100644
+--- a/src/CMakeLists.txt
++++ b/src/CMakeLists.txt
+@@ -26,6 +29,5 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
+   target_include_directories(
+     yomm2 INTERFACE
+     $<BUILD_INTERFACE:${YOMM2_SOURCE_DIR}/include>
+     $<INSTALL_INTERFACE:include>
```
Comment (Adela0814) [inline ports/yomm2/fix_find_boost.patch:35]: Why did you remove this?

File: ports/yomm2/portfile.cmake [hunk 2]
```diff
@@ -1,17 +1,29 @@
...
 vcpkg_cmake_install()
 vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/YOMM2)
-file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+if(VCPKG_LIBRARY_LINKAGE MATCHES "static") # header only library in static build
+    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib" "${CURRENT_PACKAGES_DIR}/debug")
+endif()
+
+file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```
Comment (Adela0814) [inline ports/yomm2/portfile.cmake:29]: vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Comment by Adela0814** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: ports/yomm2/fix_find_boost.patch (hunk 1 above)
Comment (FabienPean) [reply ports/yomm2/fix_find_boost.patch:35]: Do you mean `Boost::yomm2` ? It is not a valid target.

File: ports/yomm2/portfile.cmake (hunk 2 above)
Comment (FabienPean) [inline ports/yomm2/portfile.cmake:29]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
```

File: ports/yomm2/fix_find_boost.patch [hunk 3]
```diff
@@ -20,7 +20,18 @@ index 7272504..6423a17 100644
  
  # Tell library users about the Boost dependency
 -find_dependency(Boost 1.53 HINTS ${CMAKE_CURRENT_LIST_DIR}/..)
-+find_dependency(Boost 1.53)
++find_dependency(Boost 1.74 REQUIRED)
```
Comment (Adela0814) [inline ports/yomm2/fix_find_boost.patch:23]: Please note that he latest `boost` version is 1.83. Do you want to update this？

**Review by Adela0814 — APPROVED** [review]

File: ports/yomm2/portfile.cmake [hunk 4]
```diff
@@ -1,17 +1,29 @@
...
+        -DYOMM2_SHARED=${YOMM2_SHARED}
 )
 vcpkg_cmake_install()
 vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/YOMM2)
-file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+if(VCPKG_LIBRARY_LINKAGE MATCHES "static") # header only library in static build
+    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib" "${CURRENT_PACKAGES_DIR}/debug")
+endif()
```
Comment (dg0yt) [inline ports/yomm2/portfile.cmake:27]: This is confusing. Is it really header-only? Why is `lib` removed for static builds but not for dynamic builds?
To add even more confusion, x64-windows (dynamic) has a lib but not a DLL.

Comment (FabienPean) [reply ports/yomm2/portfile.cmake:27]: If not built as a shared library, it will be installed as a header-only library by default. See [CMakeLists](https://github.com/jll63/yomm2/blob/master/src/CMakeLists.txt)

Also in the README:
> By default, YOMM2 is installed as a headers only
> [...]
> The runtime can also be built and installed as a shared library, by adding -DYOMM2_SHARED=1 to the cmake command line.

I tested x64-linux and x64-linux-dynamic using a codespace. Haven't looked at Windows install. I am surprised no DLL are generated as it seems dllexport seem to be present:
https://github.com/jll63/yomm2/blob/140378a808000321d71a19ca476cda0a909a5026/src/yomm2.cpp#L6-L11

Comment (jll63) [reply ports/yomm2/portfile.cmake:27]: I introduced headers-only and I made it the default to get YOMM2 on Compiler Explorer. Building a static lib did not seem to make sense afterwards. A shared lib makes sense, so I kept it, and intend to do so in the future.

By the way, thank you @FabienPean for the vcpkg.

I have the impression that auto-installing Boost is a bit of a nuisance. I am thinking of making it an opt-in.

Comment (dg0yt) [reply ports/yomm2/portfile.cmake:27]: The static header-only build could skip the debug build entirely:
~~~
if(VCPKG_LIBRARY_LINKAGE MATCHES "static")
    set(VCPKG_BUILD_TYPE release) # header-only
endif()
~~~
at the beginning.

The dllexport is defined in a cpp file. This is at least unusual and maybe doesn't properly interact with the dllimport in the header files.
